### ISSUE TITLE
Some load specs don't generate any load

### ISSUE THREAD

**Issue by alco** [klarna/ponos#3]
Take this simple example:

```erlang
Args = [
  {name, unique_name},
  {task, fun() -> io:format("hi~n", []) end,
  {load_spec, ponos_load_specs:make_sawtooth(1, 10)},
  {options: [{duration, 10000}]}
],
ponos:add_load_generators([Args]),
ponos:init_load_generators().
```

This doesn't produce any output. I see this behaviour with `make_sawtooth` and `make_triangle`. What happens is those functions generate `0.0` intensity the first time, which prevents any future calls (due to [this condition](https://github.com/klarna/ponos/blob/e62beefc081d65c3ae7ee766672f2123f0b7d430/src/ponos_load_generator.erl#L304-L305), I assume).

**Comment by jool** [comment]
And right you are. Though it's rather due to the fact that [this](https://github.com/klarna/ponos/blob/e62beefc081d65c3ae7ee766672f2123f0b7d430/src/ponos_load_generator.erl#L336) call seems to have vanished from skip. I'll try to get around fixing this tomorrow.

**Comment by alco** [comment]
Hey, just wanted to ask, is there any progress on this?

**Comment by jool** [comment]
Sure is, sorry for the delay. I have a fix suggestion in https://github.com/klarna/ponos/tree/fix-intensity-update. Please try it out and see if it solves the issues you are seeing.

**Comment by alco** [comment]
Hey. I tried it, looks like the fix is working. Thanks!

**Comment by jool** [comment]
Solved in 1.4.0.
